{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[]},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"code","execution_count":null,"metadata":{"id":"zKeErj6SCY7d","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1764173408964,"user_tz":300,"elapsed":33616,"user":{"displayName":"Manpreet S Katari","userId":"08612708129144880191"}},"outputId":"050fa94d-9652-4cbc-93f8-b4e09c1ad47f"},"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting biopython\n","  Downloading biopython-1.86-cp312-cp312-manylinux2014_x86_64.manylinux_2_17_x86_64.manylinux_2_28_x86_64.whl.metadata (13 kB)\n","Requirement already satisfied: numpy in /usr/local/lib/python3.12/dist-packages (from biopython) (2.0.2)\n","Downloading biopython-1.86-cp312-cp312-manylinux2014_x86_64.manylinux_2_17_x86_64.manylinux_2_28_x86_64.whl (3.2 MB)\n","\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.2/3.2 MB\u001b[0m \u001b[31m16.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hInstalling collected packages: biopython\n","Successfully installed biopython-1.86\n","Reading package lists... Done\n","Building dependency tree... Done\n","Reading state information... Done\n","The following additional packages will be installed:\n","  libargtable2-0\n","The following NEW packages will be installed:\n","  clustalo libargtable2-0\n","0 upgraded, 2 newly installed, 0 to remove and 41 not upgraded.\n","Need to get 273 kB of archives.\n","After this operation, 694 kB of additional disk space will be used.\n","Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libargtable2-0 amd64 13-1.1 [14.1 kB]\n","Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 clustalo amd64 1.2.4-7 [259 kB]\n","Fetched 273 kB in 1s (369 kB/s)\n","Selecting previously unselected package libargtable2-0.\n","(Reading database ... 121713 files and directories currently installed.)\n","Preparing to unpack .../libargtable2-0_13-1.1_amd64.deb ...\n","Unpacking libargtable2-0 (13-1.1) ...\n","Selecting previously unselected package clustalo.\n","Preparing to unpack .../clustalo_1.2.4-7_amd64.deb ...\n","Unpacking clustalo (1.2.4-7) ...\n","Setting up libargtable2-0 (13-1.1) ...\n","Setting up clustalo (1.2.4-7) ...\n","Processing triggers for libc-bin (2.35-0ubuntu3.8) ...\n","/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtcm.so.1 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libumf.so.1 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libhwloc.so.15 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero_v2.so.0 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtcm_debug.so.1 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n","\n","/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n","\n","Processing triggers for man-db (2.10.2-1) ...\n","Reading package lists... Done\n","Building dependency tree... Done\n","Reading state information... Done\n","The following NEW packages will be installed:\n","  muscle\n","0 upgraded, 1 newly installed, 0 to remove and 41 not upgraded.\n","Need to get 244 kB of archives.\n","After this operation, 709 kB of additional disk space will be used.\n","Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 muscle amd64 1:3.8.1551-2build1 [244 kB]\n","Fetched 244 kB in 1s (345 kB/s)\n","Selecting previously unselected package muscle.\n","(Reading database ... 121727 files and directories currently installed.)\n","Preparing to unpack .../muscle_1%3a3.8.1551-2build1_amd64.deb ...\n","Unpacking muscle (1:3.8.1551-2build1) ...\n","Setting up muscle (1:3.8.1551-2build1) ...\n","Processing triggers for man-db (2.10.2-1) ...\n"]}],"source":["!pip install biopython\n","!apt-get install -y clustalo\n","!apt-get install -y muscle\n"]},{"cell_type":"code","source":[],"metadata":{"id":"QiJyQpR2iCxe"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["from Bio.Align.Applications import ClustalwCommandline, ClustalOmegaCommandline\n","from Bio import AlignIO\n","from Bio.Phylo.TreeConstruction import DistanceCalculator, DistanceTreeConstructor\n","from Bio import Phylo\n","import matplotlib.pyplot as plt\n","\n","# --- Run ClustalW (requires clustalw binary installed in environment) ---\n","# clustalw_cline = ClustalwCommandline(\"clustalw\", infile=\"sequences.fasta\")\n","# stdout, stderr = clustalw_cline()\n","\n","# --- Run Clustal Omega (requires clustalo binary installed in environment) ---\n","clustalomega_cline = ClustalOmegaCommandline(infile=\"/content/sequence.fasta\",\n","                                             outfile=\"/content/aligned.fasta\",\n","                                             verbose=True,\n","                                             auto=True,\n","                                             force=True)\n","stdout, stderr = clustalomega_cline()\n","\n","# --- Load alignment ---\n","alignment = AlignIO.read(\"aligned.fasta\", \"fasta\")\n","print(alignment)\n","\n","# --- Visualize alignment ---\n","fig, ax = plt.subplots(figsize=(12, 6))\n","ax.imshow([[ord(c) for c in record.seq] for record in alignment],\n","          aspect=\"auto\", cmap=\"viridis\")\n","ax.set_title(\"Clustal Alignment\")\n","plt.show()\n","\n","# --- Build phylogenetic tree ---\n","calculator = DistanceCalculator('identity')\n","dm = calculator.get_distance(alignment)\n","\n","constructor = DistanceTreeConstructor(calculator, 'nj')\n","tree = constructor.build_tree(alignment)\n","\n","Phylo.draw(tree)\n"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":383},"id":"csZfjy5IhTco","executionInfo":{"status":"error","timestamp":1764173431387,"user_tz":300,"elapsed":66,"user":{"displayName":"Manpreet S Katari","userId":"08612708129144880191"}},"outputId":"1f0d8fd2-4b0c-4165-afa3-a133de20fe0d"},"execution_count":null,"outputs":[{"output_type":"error","ename":"ModuleNotFoundError","evalue":"No module named 'Bio.Align.Applications'","traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)","\u001b[0;32m/tmp/ipython-input-1660464442.py\u001b[0m in \u001b[0;36m<cell line: 0>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32mfrom\u001b[0m \u001b[0mBio\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mAlign\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mApplications\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mClustalwCommandline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mClustalOmegaCommandline\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mBio\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mAlignIO\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mBio\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mPhylo\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTreeConstruction\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mDistanceCalculator\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mDistanceTreeConstructor\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mBio\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mPhylo\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mmatplotlib\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpyplot\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mplt\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'Bio.Align.Applications'","","\u001b[0;31m---------------------------------------------------------------------------\u001b[0;32m\nNOTE: If your import is failing due to a missing package, you can\nmanually install dependencies using either !pip or !apt.\n\nTo view examples of installing some common dependencies, click the\n\"Open Examples\" button below.\n\u001b[0;31m---------------------------------------------------------------------------\u001b[0m\n"],"errorDetails":{"actions":[{"action":"open_url","actionText":"Open Examples","url":"/notebooks/snippets/importing_libraries.ipynb"}]}}]},{"cell_type":"code","source":[],"metadata":{"id":"zHFWnAOz4K9W"},"execution_count":null,"outputs":[]}]}